package com.magneto;

import cn.hutool.core.io.FileUtil;
import org.junit.Test;
import org.zeroturnaround.zip.ZipException;
import org.zeroturnaround.zip.ZipUtil;
import java.io.File;
import java.io.FileNotFoundException;

import static org.junit.Assert.assertTrue;
import static org.junit.Assert.fail;

public class CVE_2018_1002201_Testcase1 {

    @Test(timeout = 60000)
    public void testUnpackDoesntLeaveTarget() throws Exception {
        File file = File.createTempFile("temp", null);
        File tmpDir = file.getParentFile();
        File badFile = FileUtil.file(System.getProperty("user.dir"), "src/main/resources/zip-malicious-traversal.zip");
        try {
            ZipUtil.unpack(badFile, tmpDir);
            fail("fail the test");
        } catch (ZipException e) {
            validateThrow(e);
        }
    }

    public void validateThrow(Throwable e) {
        if (e instanceof FileNotFoundException) {
            return;
        }

        assertTrue(e instanceof ZipException);
    }
}
